--- hdparm-4.1/hdparm.c~	Fri Mar  2 22:01:44 2001
+++ hdparm-4.1/hdparm.c	Sat Dec 15 19:56:25 2001
@@ -173,6 +173,7 @@
 	int i;
 	char pmodes[64] = {0,}, dmodes[128]={0,};
 	unsigned short int *id_regs=(void*)id;
+	int capacity;
 
 	printf("\n Model=%.40s, FwRev=%.8s, SerialNo=%.20s", 
 		id->model, id->fw_rev, id->serial_no);
@@ -204,14 +205,13 @@
 	}
 	if (!(id->field_valid&1))
 		printf(" (maybe):");
-	printf(" CurCHS=%d/%d/%d, CurSects=%d",
-		id->cur_cyls, id->cur_heads, id->cur_sectors,
 #ifdef BIG_ENDIAN
-		(id->cur_capacity0 << 16) | id->cur_capacity1
+	capacity = (id->cur_capacity0 << 16) | id->cur_capacity1;
 #else
-		(id->cur_capacity1 << 16) | id->cur_capacity0
+	capacity = (id->cur_capacity1 << 16) | id->cur_capacity0;
 #endif
-		);
+	printf(" CurCHS=%d/%d/%d, CurSects=%d",
+		id->cur_cyls, id->cur_heads, id->cur_sectors, capacity);
 	printf(", LBA=%s", YN(id->capability&2));
 	if (id->capability&2)
  		printf(", LBAsects=%d", id->lba_capacity);
