--- hdparm-7.7/sgio.c.orig	2007-05-14 17:12:14.000000000 +0200
+++ hdparm-7.7/sgio.c	2008-01-20 23:20:49.689196378 +0100
@@ -90,9 +90,9 @@ __u64 tf_to_lba (struct ata_tf *tf)
 	__u32 lba24, lbah;
 	__u64 lba64;
 
-	lba24 = (tf->lob.lbah << 16) | (tf->lob.lbam << 8) | (tf->lob.lbah);
+	lba24 = (tf->lob.lbah << 16) | (tf->lob.lbam << 8) | (tf->lob.lbal);
 	if (tf->is_lba48)
-		lbah = (tf->hob.lbah << 16) | (tf->hob.lbam << 8) | (tf->hob.lbah);
+		lbah = (tf->hob.lbah << 16) | (tf->hob.lbam << 8) | (tf->hob.lbal);
 	else
 		lbah = (tf->dev & 0x0f);
 	lba64 = (((__u64)lbah) << 24) | (__u64)lba24;
