--- hdparm-3.3/hdparm.c.~1~	Tue Jul 30 18:16:03 1996
+++ hdparm-3.3/hdparm.c	Tue Apr 21 22:44:57 1998
@@ -124,7 +124,13 @@
 	if (!(id->field_valid&1))
 		printf(" (maybe):");
 	printf(" CurCHS=%d/%d/%d, CurSects=%d",
-		id->cur_cyls, id->cur_heads, id->cur_sectors, *(int *)&id->cur_capacity0);
+		id->cur_cyls, id->cur_heads, id->cur_sectors,
+#ifdef BIG_ENDIAN
+	       (id->cur_capacity0 << 16) | id->cur_capacity1
+#else /* LITTLE_ENDIAN */
+	       (id->cur_capacity1 << 16) | id->cur_capacity0
+#endif
+	       );
 	printf(", LBA=%s", YN(id->capability&2));
 	if (id->capability&2)
 		printf(", LBAsects=%d", id->lba_capacity);

