diff -ru hdparm-3.6/Makefile hdparm-3.6.AX/Makefile
--- hdparm-3.6/Makefile	Fri Oct 15 14:59:00 1999
+++ hdparm-3.6.AX/Makefile	Sat Dec 25 14:10:45 1999
@@ -9,7 +9,7 @@
 all: hdparm
 
 hdparm: hdparm.c /usr/include/linux/hdreg.h
-	gcc -Wall -Wstrict-prototypes -O2 -s -o hdparm hdparm.c
+	gcc -Wall -Wstrict-prototypes $(OPTFLAGS) -o hdparm hdparm.c
  
 install: all hdparm.8
 	if [ -f $(destdir)/usr/sbin/hdparm ]; then rm -f $(destdir)/usr/sbin/hdparm ; fi 
