diff -urN hdparm-3.1-orig/Makefile hdparm-3.1/Makefile
--- hdparm-3.1-orig/Makefile	Sat Jul 27 19:07:17 1996
+++ hdparm-3.1/Makefile	Thu Sep 24 12:32:32 1998
@@ -1,9 +1,10 @@
 # Makefile for hdparm
+OPTFLAGS = -O2
  
 all: hdparm
 
 hdparm: hdparm.c /usr/include/linux/hdreg.h
-	gcc -Wall -Wstrict-prototypes -O2 -s -o hdparm hdparm.c
+	gcc -Wall -Wstrict-prototypes $(OPTFLAGS) -o hdparm hdparm.c
  
 install: all hdparm.8
 	if [ -f /usr/sbin/hdparm ]; then rm -f /usr/sbin/hdparm ; fi 
